ethereum/core/src/main/java/tech/pegasys/teku/core/signatures/record/ValidatorSigningRecord.java

```diff
@@ -92,24 +139,22 @@ public Bytes toBytes() {
   }
 
   private boolean isSafeSourceEpoch(final UInt64 sourceEpoch) {
-    return attestationSourceEpoch.equals(NEVER_SIGNED)
-        || attestationSourceEpoch.compareTo(sourceEpoch) <= 0;
+    return attestationSourceEpoch == null || attestationSourceEpoch.compareTo(sourceEpoch) <= 0;
   }
 
   private boolean isSafeTargetEpoch(final UInt64 targetEpoch) {
-    return attestationTargetEpoch.equals(NEVER_SIGNED)
-        || attestationTargetEpoch.compareTo(targetEpoch) < 0;
+    return attestationTargetEpoch == null || attestationTargetEpoch.compareTo(targetEpoch) < 0;
   }
 
-  UInt64 getBlockSlot() {
+  public UInt64 getBlockSlot() {
     return blockSlot;
   }
 
-  UInt64 getAttestationSourceEpoch() {
+  public UInt64 getAttestationSourceEpoch() {
     return attestationSourceEpoch;
   }
 
-  UInt64 getAttestationTargetEpoch() {
+  public UInt64 getAttestationTargetEpoch() {
     return attestationTargetEpoch;
   }
```

Comment:
These shouldn't be public as we want to encapsulate the data as much as we can and just have the `maySign*` methods as the key public API.  They do appear to be unused now though.